pkg/registry/apis/datasource/sub_query.go

```diff
@@ -51,8 +57,19 @@ func (r *subQueryREST) NewConnectOptions() (runtime.Object, bool, string) {
 func (r *subQueryREST) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
 	pluginCtx, err := r.builder.getPluginContext(ctx, name)
 	if err != nil {
+		if errors.Is(err, datasources.ErrDataSourceNotFound) {
+			return nil, k8serrors.NewNotFound(
+				schema.GroupResource{
+					Group:    cloudconfig.GROUP,
```

Comment:
i like this, but... is `cloudconfig.GROUP` the correct value? shouldnt we have our own "group"-thing?

Reply:
yeah this also felt off to me... I've changed it to use the current group/resource. It's a bit weird because I wasn't sure if it makes more sense to say hey cloud config couldnt find your datasource or if it should be more like prometheus couldn't find a connect because its sort of indirect you know? But anyway this looks nicer? For now I'm not sure it matters

Reply:
hmm.. not sure either...
i THINK, considering this is a `not found`, from a kubernetes perspective it should probably say what kind of thing was not found? but like.. a datasource was not found, and i don't think datasources have a k8s represeantation... but yeah, probably doesn't matter for now.